### ISSUE TITLE
[no-bare-strings] Order in `whitelist` matters

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#903]
Having:

```javascript
module.exports = {
  rules: {
    'no-bare-strings': {
      whitelist: ['&', '&times;'],
```

Will show an error when it sees `&times;` because the [implementation](https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/rules/lint-no-bare-strings.js#L236) will first remove the `&` and then won't match `&times;`.

This is *extremely* confusing.

As a side note - can you take a look at [this](https://discordapp.com/channels/480462759797063690/483601670685720591/640943548442869820) thread in Discord and tell me what was up there? I literally changed the whole file and when I ran `ember server`, something else was executed instead. I really don't understand what's going on and find this crazy. In the end I used the `./node_modules/ember-template-lint/bin/ember-template-lint.js` tool which actually *did* read the modified files and I could debug the issue. I spent hours on this and still am at a loss what is going on. Any insight will be *much* appreciated.

**Comment by rwjblue** [comment]
FWIW, it should try to find the "longest match" first. We should fix the implementation to do that (this will solve most of the scenarios where ordering matters).

**Comment by boris-petrov** [comment]
@rwjblue - yes, I was thinking that a solution is to just sort the whitelist by length, longest first. I think that will solve the issue.

If you can also take a look at the other point, that would be great.

**Comment by rwjblue** [comment]
The thing you were seeing (in the help thread) is due to using `ember-cli-template-lint` which caches the failures to redisplay without recalculating.

**Comment by boris-petrov** [comment]
Ah, OK, this begins to explain stuff. Where on the file system is this cache?

**Comment by lifeart** [comment]
in theory, https://github.com/ember-template-lint/ember-cli-template-lint/blob/master/broccoli-template-linter.js#L61

**Comment by boris-petrov** [comment]
I see. Because I literally grepped my whole file system trying to find where that cache was and couldn't. Well, no matter, I'm glad I'm not crazy. :)

Thanks for the support and for fixing this!
